block/retriever_store.go

```diff
@@ -0,0 +1,228 @@
+package block
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+
+	"github.com/evstack/ev-node/types"
+)
+
+// HeaderStoreRetrieveLoop is responsible for retrieving headers from the Header Store.
+// It retrieves both header and corresponding data before sending to heightInCh for validation.
+func (m *Manager) HeaderStoreRetrieveLoop(ctx context.Context) {
+	// height is always > 0
+	initialHeight, err := m.store.Height(ctx)
+	if err != nil {
+		m.logger.Error().Err(err).Msg("failed to get initial store height for HeaderStoreRetrieveLoop")
+		return
+	}
+	lastHeaderStoreHeight := initialHeight
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		case <-m.headerStoreCh:
+		}
+		headerStoreHeight := m.headerStore.Height()
+		if headerStoreHeight > lastHeaderStoreHeight {
+			m.processHeaderStoreRange(ctx, lastHeaderStoreHeight+1, headerStoreHeight)
+		}
+		lastHeaderStoreHeight = headerStoreHeight
+	}
+}
+
+// DataStoreRetrieveLoop is responsible for retrieving data from the Data Store.
+// It retrieves both data and corresponding header before sending to heightInCh for validation.
+func (m *Manager) DataStoreRetrieveLoop(ctx context.Context) {
```

Comment:
Same comment as with `HeaderStoreRetrieveLoop` on the error handling. It would be good to have some standard.